pallets/automation-time/src/lib.rs

```diff
@@ -1096,30 +1104,57 @@ pub mod pallet {
 			task: &TaskOf<T>,
 		) -> (Weight, Option<DispatchError>) {
 			// TODO: Handle edge case where user has enough funds to run task but not reschedule
+			if !T::DelegatorActions::is_delegation_exist(&delegator, &collator) {
+				let e = sp_runtime::DispatchErrorWithPostInfo::from(Error::<T>::DelegationNonFound);
+				Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+					task_id,
+					error_message: Into::<&str>::into(e).as_bytes().to_vec(),
```

Comment:
Discussed with Charlse. In this line of code, error_message doesn’t contain more info than the `error` so we need to pass in variables to make it more meaningful. Let’s not bother the length for now. We can prune them later.

@v9n, there are two ways to go with this `error_message`,
First, we can add full information here, e.g. "delegator:<delegator_wallet_address>. collator: <collator_wallet_address>", however, the two wallet information can be inferred and don’t really needed in this case, because delegator is the wallet who created this task and collator is specified during task creation.

Adding the full error_message will help live debugging, but as you said will increase output and maybe blockchain data size. What do you recommend, should we log the full message or leave it blank? Could leaving it blank really save block space (for example, maybe the size of the event is decided by its largest parameter)?

Reply:
@imstar15, as Vinh said, let’s add the variables to error_message in the below format, and update the tests accordingly.
"delegator:<delegator_wallet_address>, collator: <collator_wallet_address>"